src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -57,7 +57,7 @@ public override BoundNode VisitRecursivePattern(BoundRecursivePattern node)
 
         public override BoundNode VisitConstantPattern(BoundConstantPattern node)
         {
-            Visit(node.Value);
+            VisitRvalue(node.Value);
```

Comment:
I don't understand this change?

Reply:
This is something I should have changed at the same time that I made bool constants put the walker into a conditional state. Without this change, we crash in test `FlowTests.IsBoolConstant_03` because we enter a split state after visiting one of the binary pattern operands, and then attempt to visit the other binary pattern operand while in a split state.